### ISSUE TITLE
quirk of the current root/focus formulation: an unfortunate inference?

### ISSUE THREAD

**Issue by flipz357** [amrisi/amr-guidelines#275]
I think there may be a quirk in how the root is currently defined in triple form. Currently, the AMR

`(b / boy)` would read

```
root(b) ^                      /* b is the focus of the sentence */
instance(b, boy) ^         /* b is an instance of boy */
```

and `(b / bot-999)` would read

```
root(b) ^                      /* b is the focus of the sentence */
instance(b, bot-999) ^         /* b is an instance of bot-999*/
```

Imo with this there is a slightly awkward inference problem coming up here. Both graphs contain ∃b.[root(b)] which means that we can infer that they have the same root. This is true and okay. *However*, the two graphs don't have the same focus, or do they? This is now not accessible and lost. Hope I got across well enough what I mean.

I think this can be fixed by using more explicitly a "real triple" for the root/focus thing. E.g.:

```
focus(b, boy) ^             /* b focuses on boy */  (note that this also implies b is the root)
instance(b, boy) ^         /* b is an instance of boy */
```

and 

```
focus(b, bot-999) ^             /* b focuses on bot-999 */  (note that this also implies b is the root)
instance(b, bot-999)          /* b is an instance of bot-999*/
```

This would disallow the awkward inference with the matching root but mismatching focus. Indeed, with the suggested formulation, the two AMRs are completely distinct (as they should be, I think). This triple formulation would also imo align a bit better with what the AMR tries to define as "focus" (in the focus section).

**Comment by nschneid** [comment]
`b` is just a variable that refers to a specific node in the graph. Comparing two graphs requires an alignment between variables, because the variable name is understood to be arbitrary. So the two `root(b)` expressions are only judged to mean the same thing if `boy` is aligned to `bot-999`, i.e. they reflect the "same" underlying concept. Likewise, if one graph had `root(b)` while the other had `root(b2)` but those variables were aligned, they would have the same focus.

Does that make sense?

**Comment by flipz357** [comment]
Yes, that's totally clear, but it is not exactly what I was referring to. Maybe I should've used different variables to avoid this confusion. Indeed, the variables might also be named x or y, or whatever, but it's the same issue. 

E.g., the same problem would occur with the graphs

(a / b)

(z / w)

if we set a=z=x, then both allow the inference of ∃x.root(x). So then they have the same root... but they still have a different focus... The guidelines, however, kind of indicate that the root is the focus.

Hope it is now more clear what I mean with the "awkward inference" (currently we can infer that two graphs have the same root but not the same focus, which doesn't quite feel right).

[edit: minor re-wording, typo]

**Comment by nschneid** [comment]
I would interpret what you're saying as, if the root concepts don't match the graphs shouldn't be considered to have the same focus. But another way to get that result is to say the nodes shouldn't be aligned.

**Comment by nschneid** [comment]
To put it another way: Take the sentence "The tree is green." If one graph was `(g / green :domain (t / "tree"))` and the other was `(g / green-02 :ARG1 (t / "tree"))`, would you want to say they have the same focus? It depends on whether the two `g` nodes are sufficiently "similar" to align.

**Comment by flipz357** [comment]
Yes, now I think you got the correct interpretation of what I was trying to express.

> It depends on whether the two g nodes are sufficiently "similar" to align.

Indeed, but regardless of their similarity, if we align them, we get one more match (root(g)), so there is an incentive to align them that seems very independent of their actual similarity. So I think this is where the rub lies.

Judging on the similarity of nodes like green and green-01 requires background knowledge that may not always be accessible. From the point of a symbolic structure, I'd say they are even completely distinct. I'd say that the only *symbolic* similarity between your AMR examples is 1. they have the same root; 2. There is an entity that is a "tree". All other parts are different, including also that they don't have the same focus. 

Maybe I want to say that the "root" is more a structural property of the graph, and less so of the AMR. Whereas the "focus" is more a property of the AMR, but the focus is only implicitly expressed in the Penman-graph (knowing the AMR "background knowledge" that root holds the focus). So I thought that making the focus more explicit in the AMR-graph in its triple formulation could avoid unclear inferences / reduce ambiguity.

**Comment by nschneid** [comment]
My recollection from when we designed the guidelines was that we were thinking that the linguistic phenomenon of focus was represented by choosing one of the nodes to be the root. We weren't thinking in terms of two different graphs with different root concepts. It seems to me evaluating sameness-of-focus is a matter for the metrics/algorithms rather than the formalization of the graph. But maybe @uhermjakob or @cbonial or @jbonn on behalf of UMR has thoughts.

**Comment by flipz357** [comment]
For additional information, the AMR guideline currently says:

*The concept of focus only applies at the very top (root) of the AMR. After a **root concept** is selected, there are no more focus considerations -- all else is driven strictly by semantic relations. For example, once we have selected **c / college** at the root(...)*

(bold highlighting by me).

To me this seems to suggest that the phenomenon of focus is not only expressed by the actual root node, but also (or even more so?) by its label/concept. And I think this simply can be modeled with a triple `focus(x, c)` or `root(x, c)`, where `x` is the variable and `c` the concept, to be read maybe as "x/c is the focus/root" (just as it is written in above guideline excerpt). Imo using only `root(x)` would not be enough, since this would then imply that (under some alignment) every AMR has the same focus (since every AMR has a root), which may be slightly awkward.
